clientv3/watch.go

```diff
@@ -489,19 +497,31 @@ func (w *watchGrpcStream) dispatchEvent(pbresp *pb.WatchResponse) bool {
 	w.mu.RLock()
 	defer w.mu.RUnlock()
 	ws, ok := w.streams[pbresp.WatchId]
+	if !ok {
+		return false
+	}
+	if pbresp.Created {
+		// ignore created event if the watcher is not requested it
+		// or we already sent the initial created event (when we are on
+		// the resume path).
+		if !ws.initReq.createdNotify || ws.lastRev != 0 {
```

Comment:
`w.mu` doesn't protect individual `watchStream`s and `recvc` could buffer multiple create events. Probably should just be `if pbresp.Created && !ws.initReq.createdNotify { return true }` here and a `lastRev+Created` check under `<-ws.recvc` in `serveStream`.